### PR TITLE
[#6011] Add Rails 6.1 application config

### PR DESCRIPTION
**Pull request by gbp**
## Relevant issue(s)

Connected to #6011 

## What does this do?

1. Reorder configuration so all custom config is at the bottom of environment files
2. Exclude app configuration from line length and quote style Rubocop cops 
3. Regenrated app config for Rails 6.0 to reduce diff for 4.
4. Generate app config for Rails 6.1

## Why was this needed?

Framework modernisation

### CODE REVIEW

File: config/environments/test.rb [hunk 1]
```diff
@@ -40,11 +42,26 @@
...
+  #   config.action_view.raise_on_missing_translations = true
+  # end
+
+  # Annotate rendered view with file names.
+  # config.action_view.annotate_rendered_view_with_filenames = true
 
-  # Enable locale fallbacks for I18n (makes lookups for any locale fall back to
-  # the I18n.default_locale when a translation cannot be found).
+  # Enable locale fallbacks for I18n (makes lookups for any locale fall back
+  # to he I18n.default_locale when a translation cannot be found).
```
Comment (garethrees) [inline config/environments/test.rb:64]: to "he"?

File: config/initializers/filter_parameter_logging.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 # Be sure to restart your server when you modify this file.
 
 # Configure sensitive parameters which will be filtered from the log file.
-Rails.application.config.filter_parameters += [:password]
+Rails.application.config.filter_parameters += [
+  :passw, :secret, :token, :_key, :crypt, :salt, :certificate, :otp, :ssn
```
Comment (garethrees) [inline config/initializers/filter_parameter_logging.rb:5]: `:passw`? Typo or is this matching on start of word or sthn?

**Review by garethrees — COMMENTED** [review]
Had a quick skim through and looks great other than a couple of comments to double-check.

File: config/initializers/filter_parameter_logging.rb (hunk 2 above)
Comment (garethrees) [reply config/initializers/filter_parameter_logging.rb:5]: Also might want to check `token` since we might actually want that from `PostRedirect`?

Comment (gbp) [reply config/initializers/filter_parameter_logging.rb:5]: Ah yeah I was going to go back to check that, that is the new default from Rails. I guess they are matching start of params keys. Will check

File: config/environments/test.rb (hunk 1 above)
Comment (gbp) [reply config/environments/test.rb:64]: Oops

File: config/environments/development.rb [hunk 3]
```diff
@@ -69,6 +57,18 @@
   # routes, locales, etc. This feature depends on the listen gem.
   config.file_watcher = ActiveSupport::EventedFileUpdateChecker
 
+  config.action_mailer.preview_path = Rails.root.join(
```
[the remark this answers is not in the document]
Comment (gbp) [reply config/environments/development.rb:90]: Added in d810e3de4

File: config/environments/test.rb (hunk 1 above)
Comment (gbp) [reply config/environments/test.rb]: Fixed 6f1543dad

File: config/initializers/filter_parameter_logging.rb (hunk 2 above)
Comment (gbp) [reply config/initializers/filter_parameter_logging.rb:5]: Before:
```
Processing by Users::SessionsController#create as HTML
  Parameters: {"utf8"=>"✓", "authenticity_token"=>"", "user_signin"=>{"email"=>"bob@localhost", "password"=>"[FILTERED]"}, "token"=>"jof82ifzn86eshb82c5", "modal"=>"", "commit"=>"Sign in", "locale"=>"en"}
```

After:
```
Processing by Users::SessionsController#create as HTML
  Parameters: {"utf8"=>"✓", "authenticity_token"=>"[FILTERED]", "user_signin"=>{"email"=>"bob@localhost", "password"=>"[FILTERED]"}, "token"=>"[FILTERED]", "modal"=>"", "commit"=>"Sign in", "locale"=>"en"}
```

Comment (gbp) [reply config/initializers/filter_parameter_logging.rb:5]: > Also might want to check :token

Yeah we use token other places too like project invites so I think we should drop `:token`.

Comment (gbp) [reply config/initializers/filter_parameter_logging.rb]: Updated in d16de64

**Comment by gbp** [comment]
@garethrees anything more feedback?

**Comment by garethrees** [comment]
Nope, LGTM!

Only thing I'd do on merge is update the commit message of 3371ff75f1b52b03e257fdd6e479b4e508754f0f to note that we've explicitly removed the default `:token` value from `filter_parameters`.

**Pull request merged**